src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3966,3 +4129,126 @@ void DoorLockServer::DoorLockOnAutoRelockCallback(System::Layer *, void * callba
+
+    return aEncoder.EncodeList([delegate](const auto & encoder) -> CHIP_ERROR {
+        for (uint8_t i = 0; true; i++)
+        {
+            uint8_t buffer[kAliroProtocolVersionSize];
+            MutableByteSpan protocolVersion(buffer);
+            auto err = delegate->GetAliroExpeditedTransactionSupportedProtocolVersionAtIndex(i, protocolVersion);
+            if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+            {
+                return CHIP_NO_ERROR;
+            }
+            ReturnErrorOnFailure(err);
+            ReturnErrorOnFailure(encoder.Encode(protocolVersion));
+        }
+    });
+}
+
+CHIP_ERROR DoorLockServer::ReadAliroSupportedBLEUWBProtocolVersions(const ConcreteReadAttributePath & aPath,
+                                                                    AttributeValueEncoder & aEncoder, Delegate * delegate)
+{
+    VerifyOrReturnValue(delegate != nullptr, aEncoder.EncodeEmptyList());
+
+    return aEncoder.EncodeList([delegate](const auto & encoder) -> CHIP_ERROR {
+        for (uint8_t i = 0; true; i++)
+        {
+            uint8_t buffer[kAliroProtocolVersionSize];
+            MutableByteSpan protocolVersion(buffer);
+            auto err = delegate->GetAliroSupportedBLEUWBProtocolVersionAtIndex(i, protocolVersion);
+            if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+            {
+                return CHIP_NO_ERROR;
+            }
+            ReturnErrorOnFailure(err);
+            ReturnErrorOnFailure(encoder.Encode(protocolVersion));
+        }
+    });
+}
+
+CHIP_ERROR DoorLockServer::ReadAliroByteSpanAttribute(CHIP_ERROR (Delegate::*func)(MutableByteSpan &), MutableByteSpan & data,
+                                                      Delegate * delegate, AttributeValueEncoder & aEncoder, bool isNullable)
+{
+    VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Delegate is null"));
+
+    ReturnErrorOnFailure((delegate->*func)(data));
+    if (isNullable && data.empty())
+    {
+        ReturnErrorOnFailure(aEncoder.EncodeNull());
+    }
+    else
+    {
+        ReturnErrorOnFailure(aEncoder.Encode(data));
+    }
+    return CHIP_NO_ERROR;
+}
+
+// Implements the read functionality for the AttributeAccessInterface.
+CHIP_ERROR DoorLockServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    if (aPath.mClusterId != Clusters::DoorLock::Id)
+    {
+        // We shouldn't have been called at all.
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    Delegate * delegate = GetDelegate(aPath.mEndpointId);
+
+    switch (aPath.mAttributeId)
+    {
+    case AliroReaderVerificationKey::Id: {
+        uint8_t buffer[kAliroReaderVerificationKeySize];
+        MutableByteSpan readerVerificationKey(buffer);
+        return ReadAliroByteSpanAttribute(&Delegate::GetAliroReaderVerificationKey, readerVerificationKey, delegate, aEncoder,
+                                          true);
```

Comment:
Either `/* isNullable = */ true` or better yet have an enum instead of a boolean.  It's impossible to tell at this callsite whether this `true` is correct or not without going to read the other thing.